File: backend/app/routes/crew_analyst.py
from fastapi import APIRouter
from pydantic import BaseModel
from typing import List
from app.crew.stock_analyst import StockAnalysisCrew
import json

router = APIRouter()
crew_runner = StockAnalysisCrew()

class CrewRequest(BaseModel):
    tickers: List[str]
    period: str = "1mo"
    interval: str = "1d"

@router.post("/crew/stock-analysis")
async def run_stock_analysis_crew(request: CrewRequest):
    result = crew_runner.run(
        tickers=request.tickers,
        period=request.period,
        interval=request.interval
    )

    # Extract summary and market data
    summary_text = result["summary"]
    market_data_raw = None
    for task in result["tasks_output"]:
        if task["agent"] == "Market Data Collector":
            market_data_raw = task["raw"]
            break

    # Parse market data
    try:
        market_data = json.loads(market_data_raw)
    except Exception:
        market_data = {"error": "Failed to parse market data"}

    return {
        "summary": summary_text,
        "market_data": market_data
    }

File: src/utils/api/crew-analyst.ts
// utils/api/crewStock.ts

export async function fetchCrewStockAnalysis(tickers: string[], period = "1mo", interval = "1d") {
    const response = await fetch("http://localhost:8000/api/crew/stock-analysis", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({ tickers, period, interval })
    });
  
    if (!response.ok) throw new Error("Failed to fetch stock analysis");
    return response.json();
  }
  